utils/_context/_scenarios/docker_ssi.py

```diff
@@ -142,8 +142,11 @@ def fill_context(self, json_tested_components):
 
     def post_setup(self):
         logger.stdout("--- Waiting for all traces and telemetry to be sent to test agent ---")
-        time.sleep(60)  # wait for the traces and telemetry to be sent to the test agent
-        interfaces.test_agent.collect_data(f"{self.host_log_folder}/interfaces/test_agent")
+        data = None
+        attempts = 0
+        while attempts < 30 and not data:
+            data = interfaces.test_agent.collect_data(f"{self.host_log_folder}/interfaces/test_agent")
```

Comment:
this might not be enough, since we might get a trace, but not telemetry data?

Reply:
The tests are passing on CI so it seems like it's sufficient for the moment.